types/type.go

```diff
@@ -0,0 +1,23 @@
+package types
+
+type NotificationType string
+
+const (
+	WorkitemCreate       NotificationType = "workitem.create"
+	WorkitemUpdate       NotificationType = "workitem.update"
+	CommentCreate        NotificationType = "comment.create"
+	CommentUpdate        NotificationType = "comment.update"
+	UserEmailUpdate      NotificationType = "user.email.update"
+	InvitationTeamNoorg  NotificationType = "invitation.team.noorg"
+	InvitationSpaceNoorg NotificationType = "invitation.space.noorg"
+	AnalyticsNotifyCVE   NotificationType = "analytics.notify.cve"
+)
+
+var notifiers = map[NotificationType][]string{
+	AnalyticsNotifyCVE: {"fabric8-gemini-server"},
+	UserEmailUpdate:    {"fabric8-auth"},
+}
+
+func (nType NotificationType) Notifiers() []string {
+	return notifiers[nType]
+}
```

Comment:
@aslakknutsen I am not re-factoring Template Registry [link](https://github.com/fabric8-services/fabric8-notification/blob/master/template/registry.go#L99) to use NotificationType.  Essentially, TemplateRegistry is not a registry . It just a helper thing.  This can stay independently and no need to attach with NotificationType.  I have make sure that all the hard_coding of notification_type_string is removed from TemplateRegistry and related stuff.